src/ol/source/Raster.js

```diff
@@ -815,18 +823,19 @@ class RasterSource extends ImageSource {
       if (imageData) {
         imageDatas[i] = imageData;
       } else {
-        return;
+        return Promise.resolve();
```

Comment:
Minor, but here and above it seems like we could reuse a single promise from `Promise.resolve()` created at the module level.